### PR TITLE
Optimize layers

### PR DESCRIPTION
**Pull request by LaurentGoderre**
Collpase installation layers and image went from 550Mb to 370mb

### CODE REVIEW

File: Dockerfile-debian.template [hunk 1]
```diff
@@ -72,10 +72,8 @@ RUN set -ex; \
         | sort -u \
         | xargs -rt apt-mark manual; \
     \
-    apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false
-
-# Install required libs
-RUN set -ex; \
+    apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false; \
+    # Install required libs
```
Comment (whalelines) [inline Dockerfile-debian.template:76]: Why would you move this?

File: Dockerfile-debian.template [hunk 2]
```diff
@@ -3,12 +3,12 @@ FROM php:8.2-%%VARIANT%%
 LABEL org.opencontainers.image.source=https://github.com/espocrm/espocrm
 LABEL org.opencontainers.image.description="EspoCRM is an Open Source CRM. Try for Free."
 
-# Install php libs
 RUN set -ex; \
     \
     aptMarkList="$(apt-mark showmanual)"; \
     \
     apt-get update; \
+    # Install php libs
```
Comment (whalelines) [inline Dockerfile-debian.template:11]: The whole thing is about PHP, so may not move this?

**Comment by LaurentGoderre** [comment]
@whalelines the answer to both questions is that the second install step relies on the cache not being cleared in the previous step. So either we duplicate the update and clear cache or we put them in the same layer. The latter is faster to build

**Comment by whalelines** [comment]
But if you are combining those two `RUN` steps into one, why would you run the purge step at all in the middle of it?

**Comment by LaurentGoderre** [comment]
That purge gets rids of build dependencies for the PHP deps which is not the same as the cache.

**Comment by tmachyshyn** [comment]
@LaurentGoderre thanks for your contribution

**Pull request merged**